elastic_search/index.js

```diff
@@ -0,0 +1,135 @@
+/*
+This lambda function fetched search results from elastic search instance
+
+Request parameter has two nodes, state and secret.
+`state` node is empty for the first request. In this lambda function, cursor for entity `search` is stored in `state` object.
+`secrets` node contains the host name of elastic search instance 
+{
+    "state" : {},
+    "secrets" : {
+        "host" : "host_name"
+    }
+}
+
+Insert node contains records fetched using search query `bag`.
+Delete node contains records fetched using search query `puma`.
+
+Primary key for the records are [_id, _index, _type, sync_time]
+*/
+var RecordOperation = {
+    INSERT : 1,
+    DELETE : 2
+}
+
+exports.handler = (request, context, callback) => {
+    update(request.state, request.secrets, callback)
+}
+
+async function update(state, secrets, callback) {
+    let response = initializeResponse(state);
+     
+    let insertResponse = fetchSearchResult(state, secrets, 'products', 'product', 'Bag');
+    let deleteResponse = fetchSearchResult(state, secrets, 'products', 'product', 'Puma');
+    
+    insertResponse.then(function(searchResult) {
+        putSearchResult(response, searchResult, RecordOperation.INSERT)
+    }).catch(function (err) {
+        callback(err); 
+    });
+    
+    deleteResponse.then(function(searchResult) {
+        putSearchResult(response, searchResult, RecordOperation.DELETE)
+    }).catch(function (err) {
+        callback(err); 
+    });
+    
+    await insertResponse;
+    await deleteResponse;
+    
+    response.state = {
+        "search" : new Date()
+    }
+    
+    callback(null, response);
+}
+
+function initializeResponse(state) {
+    return {
+        "state": {...state}, 
```

Comment:
I'm not sure why we enclose js object fields into quotes, it's redundant. 
But if we have this in our docs, we should be at least consistent and add quotes for `hasMore` too.
While quotes are for json format, while here it's js object declaration.

Reply:
btw, no quotes here https://github.com/fivetran/functions/pull/6/files#diff-de246c2522a0583092e324bcb864f070R74 (agree with that)